src/AppBundle/Model/EventWiki.php

```diff
@@ -321,7 +321,8 @@ protected function getPageIds(string $type): array
             ? stream_get_contents($this->$propertyName)
             : $this->$propertyName;
 
-        return explode(',', bzdecompress($blob));
+        // Decompressing the empty string or null results in an empty string, so array_filter removes this.
+        return array_filter(explode(',', bzdecompress($blob)));
```

Comment:
This is a good sanity check, but I'm wondering if we should also remove NULL and empty values when **storing** the data; we've gone to some length to minimize the stored string so to accommodate large ID pools -- it sounds like it could be a good idea to prevent "waste" of space in the storage too, to begin with?

Reply:
No, I'm pretty sure we're never actually storing empty IDs, at least I haven't been able to make it do that. We're *retrieving* an empty value — a single one — only when pulling from a NULL DB value (because `bzdecompress(null)` and `bzdecompress('')` both give us an empty string). Which means the `array_filter` is slightly unnecessary (we could instead check for empty string, but this is smaller; I don't know if there's performance differences; should I switch it?).

Reply:
Then why was the bug with (,xxx,yyy) ? That suggested that you have an array of [ null, xxx, yyy ] not just one null, no?

Reply:
Before there was only one field, `page_ids`, so `explode(',', bzdecompress(null))` gives an array with one blank value. Then in EventRepository we use `implode(',', $pageIds)` for use in SQL, whilst [ignoring empty strings](https://github.com/wikimedia/eventmetrics/blob/master/src/AppBundle/Repository/EventRepository.php#L352) (meaning there were no page IDs). With #154, however, we have two page ID fields. `EventWiki::getPages()` merges the arrays and you can end up with `['', 1, 2]`. The `implode` function will produce `,1,2`, hence the broken SQL. So we're not storing blank values for page IDs as with `[null, 1, 2]`, rather we're correctly storing `null` in one field and `bzcompress(implode(',', [1, 2]))` in the other.